pyalgostrategypool/aroon_crossover.py

```diff
@@ -0,0 +1,193 @@
+import talib
+from pyalgotrading.constants import *
+from pyalgotrading.strategy.strategy_base import StrategyBase
+
+import logging
+logger = logging.getLogger('strategy')
+
+
+class AroonCrossover(StrategyBase):
+
+    def __init__(self, *args, **kwargs):
+        """
+        Accept and sanitize all your parameters here
+        setup the variables you will need here
+        if you are running the strategy for multiple days, then this method will be called only once at the start of the strategy
+        """
+
+        super().__init__(*args, **kwargs)
+
+        # Aroon parameter
+        self.time_period = self.strategy_parameters['TIME_PERIOD']
+
+        # Sanity
+        assert (0 < self.time_period == int(self.time_period)), f"Strategy parameter TIME_PERIOD should be a positive integer. Received: {self.time_period}"
+
+        # Variables
+        self.main_order = None
+
+    def initialize(self):
+        """
+        If you are running the strategy for multiple days, then this method will be called once at the start of every day
+        use this to initialize and re-initialize your variables
+        """
+        self.main_order = {}
+
+    @staticmethod
+    def name():
+        """
+        Name of your strategy
+        """
+        return 'Aroon Crossover'
+
+    @staticmethod
+    def versions_supported():
+        """
+        Strategy should always support the latest engine version
+        Current version is 3.3.0
+        """
+        return [AlgoBullsEngineVersion.VERSION_3_3_0]
+
+    def get_decision(self, instrument):
+        """
+        This method calculates the crossover using the hist data of the instrument along with the required indicator and returns the entry/exit action
+        """
+
+        # Get OHLC historical data for the instrument
+        hist_data = self.get_historical_data(instrument)
+
+        # Calculate the Stochastic values
+        aroondown, aroonup = talib.AROON(hist_data['high'], hist_data['low'], timeperiod=self.time_period)
+
+        # Get the crossover value
+        aroon_crossover_value = self.utils.crossover(aroonup, aroondown)
+
+        if aroon_crossover_value == 1:
+            action = 'BUY'
+        elif aroon_crossover_value == -1:
+            action = 'SELL'
+        else:
+            action = None
+        return action
```

Comment:
lets print the relevant details and the action in one line everywhere

do not print hist data rows, instead give individual columns in the prints

Reply:
Print not added because crossover gets printed anyways and entry/exit is present in the order transaction type, please let me know if this is correct.